kratos/python_scripts/output_process_with_controller.py

```diff
@@ -0,0 +1,64 @@
+# Importing the Kratos Library
+import KratosMultiphysics
+
+from KratosMultiphysics.model_parameters_factory import KratosModelParametersFactory
+
+
+def Factory(settings, Model):
+    if not isinstance(settings, KratosMultiphysics.Parameters):
+        raise Exception("expected input shall be a Parameters object, encapsulating a json string")
+    return OutputProcessWithController(Model, settings["Parameters"])
+
+class OutputProcessWithController(KratosMultiphysics.OutputProcess):
+    """ This process divides its responsabilities between two objects:
+        a controller that decides whether current time step must be
+        printed or not and a print process that controls what will
+        be printed and in which format.
+    """
+    def __init__(self, model, params):
+        KratosMultiphysics.OutputProcess.__init__(self)
+        default_settings = self.GetDefaultParameters()
+        self.params = params
+        self.model = model
+        self.params.ValidateAndAssignDefaults(default_settings)
+        factory = KratosModelParametersFactory(self.model)
+        self.controller = factory.ConstructItem(self.params["controller_settings"])
+        self.print_process = factory.ConstructItem(self.params["output_settings"])
```

Comment:
```suggestion
        self.output_process = factory.ConstructItem(self.params["output_settings"])
```
Very minor in the naming of the variables for the sake of consistency (feel free to dismiss).

Reply:
done!